examples/slack-security-findings/daily_findings.py

```diff
@@ -0,0 +1,188 @@
+from __future__ import annotations
+
+import json
+import os
+import sys
+from datetime import date, timedelta
+from http.client import RemoteDisconnected
+from json import JSONDecodeError
+from string import Template
+from typing import TypedDict, Callable, Any
+from urllib import request
+from urllib.error import URLError
+import logging
+from argparse import ArgumentParser
+
+LOG = logging.getLogger(__name__)
+
+
+class Finding(TypedDict):
+    href: str
+    first_seen_snapshot_date: date
+    file_path: str
+    start_line: int
+    end_line: int
+    type: str
+    severity: str
+    severity_score: float
+    status: str
+
+
+class SigridApiClient:
+
+    def __init__(self, customer: str, system: str, token: str):
+        sigrid_api_endpoint = 'https://sigrid-says.com/rest/analysis-results/api/v1/security-findings/$customer/$system'
+        self.sigrid_api = Template(sigrid_api_endpoint).substitute(customer=customer, system=system)
+        self.token = token
+
+    def get_findings(self) -> Any | None:
+        try:
+            req = request.Request(self.sigrid_api)
+            req.add_header('Authorization', 'Bearer ' + self.token)
```

Comment:
Note this might not work if we make the token an environment variable, since environment variables are `str` and `add_header` doesn't like that.

Reply:
I meant in reverse, that `add_header` expects bytes which will fail when the token contains non-ASCII characters.

However, I was wrong and `add_header` happily supports `str` these days, probably for several years already.